via/static/scripts/video_player/hooks/use-app-layout.ts

```diff
@@ -0,0 +1,55 @@
+import type { RefObject } from 'preact';
+import { useCallback, useLayoutEffect, useState } from 'preact/hooks';
+
+/* Relative application container size */
+type AppSize = 'sm' | 'md' | 'lg' | 'xl' | '2xl';
+
+/**
+ * Return a string representing the relative size of `appContainer` that can
+ * be used to make decisions about app layout and sizing of components. This
+ * would typically be something handled by media queries, but it is an element
+ * (`body`) width that needs to be queried, not the viewport. We could replace
+ * this with container queries once browser support is sufficient.
+ */
+export function useAppLayout(appContainer: RefObject<HTMLDivElement | null>) {
```

Comment:
I suggest to explicitly specify the return type here for the reader, and also export the `AppSize` type.